src/main/resources/com/google/api/codegen/php/standalone_sample.snip

```diff
@@ -43,64 +43,95 @@
+                @if sampleFile.extraInfo.missingDefaultServiceAddress
+                    'serviceAddress' => 'my-service-address',
+                @end
+                @if sampleFile.extraInfo.missingDefaultServiceScopes
+                    'credentialsConfig' => [
+                        'scopes' => ['my-service-scope'],
+                    ],
+                @end
+                ];
+            ${@apiMethod.apiVariableName} = new {@apiMethod.apiClassName}($options);
+
+        @else
+            ${@apiMethod.apiVariableName} = new {@apiMethod.apiClassName}();
+
+        @end
+        @if sample.initCode.lines
+            {@initCode(sample.initCode, apiMethod.apiVariableName)}
+
+        @end
+        try {
+            @switch sample.callingForm
+            @case "LongRunningRequest"
+                {@longRunningSampleCode(sample, apiMethod)}
+            @case "LongRunningRequestAsync"
+                {@longRunningAsyncSampleCode(sample, apiMethod)}
+            @case "Request"
+                {@requestSampleCode(sample, apiMethod)}
+            @case "RequestPaged"
+                {@pagedSampleCode(sample, apiMethod)}
+            @case "RequestPagedAll"
+                {@pagedAllSampleCode(sample, apiMethod)}
+            @case "RequestStreamingBidi"
+                {@bidiStreamingSampleCode(sample, apiMethod)}
+            @case "RequestStreamingBidiAsync"
+                {@bidiStreamingAsyncSampleCode(sample, apiMethod)}
+            @case "RequestStreamingClient"
+                {@clientStreamingSampleCode(sample, apiMethod)}
+            @case "RequestStreamingClientAsync"
+                {@clientStreamingAsyncSampleCode(sample, apiMethod)}
+            @case "RequestStreamingServer"
+                {@serverStreamingSampleCode(sample, apiMethod)}
+            @default
+                $unhandledCallingForm: {@sample.callingForm} in sample "{@apiMethod.getClass.getSimpleName}"$
             @end
-            @if sampleFile.extraInfo.missingDefaultServiceScopes
-                'credentialsConfig' => [
-                    'scopes' => ['my-service-scope'],
-                ],
+        } finally {
+            @if sampleFile.extraInfo.hasMissingDefaultOptions
+                if (isset(${@apiMethod.apiVariableName})) {
+                    ${@apiMethod.apiVariableName}->close();
+                }
+            @else
+                ${@apiMethod.apiVariableName}->close();
             @end
-            ];
-        ${@apiMethod.apiVariableName} = new {@apiMethod.apiClassName}($options);
+        }
+    }
 
+    {@optsArray(sample.initCode.argDefaultParams)}
 
-    @else
-        ${@apiMethod.apiVariableName} = new {@apiMethod.apiClassName}();
+    $options = getopt('', $opts);
+
+    @if sample.initCode.argDefaultParams
+        @join param : sample.initCode.argDefaultParams
+            {@variable(param.identifier)} == $options['{@param.identifier}'] ?? {@renderInitValue(param.initValue, param.identifier)};
```

Comment:
`??` is not available in PHP 5.*, which we want to support :-(

You may need to use the much wordier:
```
isset($options['{@param.identifier}']) ? $options['{@param.identifier}'] : {@renderInitValue(param.initValue, param.identifier)}
```

Also, should this be `=` instead of `==`?

Reply:
Following up on the structure here - a pattern than we use elsewhere for setting default arguments is:
- Create a `$defaultOptions` array with all defaults set
- Create the `$options` array using whatever method (in this case getopts)
- `$options += $defaultOptions`

That may work here and avoids the `isset...` construction. It is very sad that we don't have the `??` operator.

Reply:
**^--- Note:** from the _Developer Experience_ perspective: I have no qualms with taking a wordier (more idiomatic & correct) approach 🙂

The CLI parsing will generally not be shown to developers, ie. it's usually not included when we embed samples in the website. Sometimes it may be, so...

So long as the generated CLI parsing & main() invoking code is _**idiomatic**_ to the language & _looks familiar_ to anyone who's used this option parser – I'm +1 from my side!